### ISSUE TITLE
Bloom something

### ISSUE THREAD

**Issue by staltz** [ssb-ngi-pointer/ssb-db2#147]
Not sure if we can/should use bloom filters, but this library looks good and supports export/import to JSON string. https://callidon.github.io/bloom-filters/

**Comment by staltz** [comment]
Like, prefix indexes are basically similar to bloom filters, because they behave the same regarding false positives. But maybe we could have a bloom filter for "I follow X" foreach X and another for "I block X" foreach X. Maaaybe. But there are also other data structures there.

**Comment by arj03** [comment]
Yes that library is very interesting. Will have a look when I get better time :)

**Comment by arj03** [comment]
I was reading up on the idea of learned index (wierd name for what it is), there is an implementation of it [here](https://github.com/learnedsystems/RadixSpline). Basically another index form that could be useful in some specific cases. From [this](https://news.ycombinator.com/item?id=25899286) HN discussion.

**Comment by staltz** [comment]
> I was reading up on the idea of learned index (wierd name for what it is), there is an implementation of it [here](https://github.com/learnedsystems/RadixSpline). Basically another index form that could be useful in some specific cases. From [this](https://news.ycombinator.com/item?id=25899286) HN discussion.

So cool.

**Comment by arj03** [comment]
I was reading up on set replication and was reading Kleppmann's [algorithm](https://martin.kleppmann.com/2020/12/02/bloom-filter-hash-graph-sync.html) for this. It has this interesting graph:

![image](https://user-images.githubusercontent.com/350682/109001698-86e88980-76a5-11eb-8bd7-1305cdf06209.png)

**Comment by arj03** [comment]
Hah, I'm trying to run his python script on 32 bits and I think there is a hidden crypto miner in there, or it's just very inefficient. There is a very big difference between 32 and 10.
